### PR TITLE
[terra-table] Screen Reader should announce state change for Rows and header checkbox.

### LINKED ISSUE #3044: [terra-table] Checkbox does not read selection state change in chrome + jaws.
# Bug Report
Checkbox does not read selection state change in chrome + jaws.

## Description
When using JAWS w/ Chome the activation of a checkbox does not read the newly updated state. It will only read "space" instead of also announcing the new state.

## Steps to Reproduce
1. Launch terra-ui.com with Chrome, navigate to terra-table's "Check Mark Table" example.
2. Launch JAWS, navigation to checkbox element within table.
3. Use space to toggle the state of the checkbox.
4. Observe state change not read.

## Expected Behavior
JAWS should read the interaction and the new checkbox state.

## Environment
* Component Name and Version: terra-table v4.9.0
* Browser Name and Version: Chrome Version 83.0.4103.116
* Operating System and version (desktop or mobile): Windows 10

## @ Mentions
@scottwilmarth

### PR DESCRIPTION
**Pull request by pranav300**
### Summary

This PR 
1. Removes `VisuallyHiddentext` from the checkbox.
2. The original tab order is maintained.
2. The Focus on the tab directly goes into the inner div `checkmark` which then programmatically transfers it to the parent div to maintain the original design.

Closes #3044 

### Deployment Link

https://terra-core-jaws-checkma-qybhqh.herokuapp.com/components/terra-table/table/guides/check-mark-table

### Testing

In Jaws + Chrome:
Selected: `space, checked`
Not Selected: `space, unchecked`

In VoiceOver + Safari
Selected: `tick, {label}, tickbox`
Not Selected: `untick, {label}, tickbox`

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-table/src/subcomponents/_CheckMarkCell.jsx [hunk 1]
```diff
@@ -134,28 +135,33 @@ const CheckMarkCell = ({
...
     { 'is-interactable': !isDisabled && isSelectable },
     { 'is-top-align': attrPadding },
+    { 'is-focused': isFocused && !isDisabled },
   );
 
   return (
     <div
       {...customProps}
       {...attrSpread}
+      tabIndex="-1"
```
Comment (dkasper-was-taken) [inline packages/terra-table/src/subcomponents/_CheckMarkCell.jsx:145]: Is this index unnecessary or are we missing the equivalent in the HeaderCheckMarkCell?

File: packages/terra-table/src/subcomponents/_HeaderCheckMarkCell.jsx [hunk 2]
```diff
@@ -112,14 +112,14 @@ const HeaderCheckMarkCell = ({
...
         onMouseDown: wrappedEventCallback(onMouseDown, event => event.currentTarget.setAttribute('data-cell-show-focus', 'false')),
-        tabIndex: '0',
         'data-cell-show-focus': true,
       };
     }
 
     attrCheck = {
       role: 'checkbox',
       'aria-checked': isSelected && isIndeterminate ? 'mixed' : isSelected,
+      tabIndex: '0',
```
Comment (dkasper-was-taken) [inline packages/terra-table/src/subcomponents/_HeaderCheckMarkCell.jsx:122]: Should this index account for isDisabled with a -1 index?

File: packages/terra-table/src/subcomponents/_CheckMarkCell.jsx (hunk 1 above)
Comment (pranav300) [reply packages/terra-table/src/subcomponents/_CheckMarkCell.jsx]: removed it here https://github.com/cerner/terra-core/pull/3126/commits/bc141da851635bf0c558673882deb0dfafdac8ca.

File: packages/terra-table/src/subcomponents/_HeaderCheckMarkCell.jsx (hunk 2 above)
Comment (pranav300) [reply packages/terra-table/src/subcomponents/_HeaderCheckMarkCell.jsx]: updated here https://github.com/cerner/terra-core/pull/3126/commits/bc141da851635bf0c558673882deb0dfafdac8ca

**Comment by emilyrohrbough** [comment]
Spoke with @PaulSolDev & since @jeremyfuksa completed the Accessibility Review, this PR is good to go. Removed the `Functional Review Required Label`.

**Pull request merged**